common/smoke-test/run.js

```diff
@@ -22,37 +26,36 @@ async function main() {
     }
 
     const skipFiles = smokeTestConfig.skip || [];
-    const jsFiles = fs.readdirSync(entry.SamplesDirectory)
+    const jsFiles = fs
+      .readdirSync(entry.SamplesDirectory)
       .filter((name) => name.endsWith(".js"))
       .filter((name) => !skipFiles.includes(name));
 
-    for (let targetSample of jsFiles) {
-      const sampleModule = require(`${entry.SamplesDirectory}/${targetSample}`);
+    for (const targetSample of jsFiles) {
       samplesToExecute.push({
-        entrypoint: sampleModule.main,
         name: entry.Name,
         sampleFile: targetSample,
-        directory: entry.SamplesDirectory
+        directory: entry.SamplesDirectory,
       });
     }
   }
 
   // Run all samples
-  for (let sample of samplesToExecute) {
-    let result = await executeSample(sample);
+  for (const sample of samplesToExecute) {
+    const result = await executeSample(sample);
 
     if (!result.success) {
-      exitCode = 1;
+      exitCode++;
```

Comment:
NIT: would it be possible to remove `exitCode` now and just call process.exit with `failures.length`? later?

Reply:
I'd be careful about doing this. Certain exit codes have conventional meanings on non-windows platforms. Technically nothing should care about it, but we probably want to always `exit(1)` and print additional info if necessary unless we have in mind to interpret different exit codes differently in our tooling.

Reply:
FWIW Mocha does this (with a cap of 255), but some folks think it is not a good idea: https://github.com/mochajs/mocha/issues/3559

They are stuck with backcompat, but perhaps we want to avoid this trap if it's not too hard to do so? Maybe instead of relying on exit codes we could return toolable output like JSON?

Reply:
I probably got the idea from mocha then.  Currently we just need a non-zero exit code to tell the pipeline that this program has failed.  I changed it back to exit with 1.